### ISSUE TITLE
Blogname appearing in meta descriptions, event when that option is disabled

### ISSUE THREAD

**Issue by kevinlisota** [sybrew/the-seo-framework#149]
In description meta settings, I have unchecked "add blogname to automated description additions" and also unchecked "add additions to automated description." We don't want either of those in our meta descriptions.

On certain archive types, when there is no description set, SEO Framework is inserting an auto-generated one that is simply two words "on Blogname"

I'm seeing this on tag and author archives, along with other taxonomies when no description is set. Date and CPT archives have a different issue that I already reported.

Your auto-generation should be completely turned off when those boxes are unchecked in the UI.

**Comment by sybrew** [comment]
What you're encountering is happening when no description is found, and when no useful content is found either. Like when using a page builder.

It's a fail-safe exception, implemented as a feature: [The code](https://github.com/sybrew/the-seo-framework/blob/1691e8718c516b2933bd3811f4a490b1457ed898/inc/classes/generate-description.class.php#L444-L449).
There's a filter to disable it regardless (see 4th parameter `$ignore`) [found here](https://github.com/sybrew/the-seo-framework/blob/1691e8718c516b2933bd3811f4a490b1457ed898/inc/classes/generate-description.class.php#L715).

I understand that you want perfect granular and expected control over the output, but please keep in mind that not everyone has such a keen eye 😄. TSF sometimes fills in the gaps automatically, sometimes when least expected.

For Author-archives, see https://github.com/sybrew/the-seo-framework/issues/61.
For special queries, see https://github.com/sybrew/the-seo-framework/issues/20.

**Comment by kevinlisota** [comment]
Thanks. I'll give that filter a try.

I'm not sure it is a great fail-safe though. Google doesn't like duplicate meta descriptions, and that is exactly what this will generate. Our site has 13,000 tags, none of which have a description set. That means our tags alone will generate 13,000 duplicate meta description errors on Google.

**Comment by sybrew** [comment]
Well, it's better than having no description. In many cases, tags are ones that shouldn't be indexed.

Regardless, if tags, categories or even pages' name collide, then you're probably mixing content anyway.
With these as a fallback, it might even be easier to spot duplicated pages 😄. So, we can see it as a win 👍

**Comment by kevinlisota** [comment]
Actually, I don't agree with that. A two word meta description is too short, and Google will pick up some other information on the page, usually the first excerpt, to fill it in. That is what it is doing on our site right now. It is ignoring the two word meta description and using an article excerpt from the top of the page.

In our case, we have no collisions between tags, category and page naming. A tag archive page never looks the same as a category archive page and are unique. We need them indexed. While the same post may appear on a category or tag archive, the assortment of posts in various categories and tags makes the content of each page unique and relevant to the tag.

This is different from the date archives. I agree that those should often not be indexed, as they may easily duplicate content in the main blog loop.

**Comment by sybrew** [comment]
> That is what it is doing on our site right now. It is ignoring the two word meta description and using an article excerpt from the top of the page.

That's true, which means it's self-resolving. Google does this regardless, even with titles, when it finds the metadata uncompelling for the query:
![image](https://cloud.githubusercontent.com/assets/7662236/26764012/f9a86616-495e-11e7-8e34-44283a29d70a.png)

In any case, you're describing an issue you're having. How would you like it to be resolved? 😮 
Disabling it through the filter is one I've provided, but if you'd have a better solution I'd love to hear it 😄.

**Comment by kevinlisota** [comment]
How can I use the $ignore parameter that you mention above in the `the_seo_framework_generated_description_additions` filter?

You pass a 2-level array, but looking at what is available in the callback, I think it is only able to alter values in the 2nd level key-value pair array. The other three parameters are just available as parameters in the callback, not as changeable values.

**Comment by sybrew** [comment]
Hi @kevinlisota,

`apply_filters_ref_array` works like `apply_filters`, but it passes a single array, instead of many variables.
Each array entry is like one of those variables.
This is also known as a variadic function, although that's only truly available from PHP 5.6.

Here's an example of the filter usage:
```
/**
 * Filters 'the_seo_framework_generated_description_additions'
 *
 * @param array $data   The description data.
 * @param int   $id     The object ID.
 * @param mixed $term   The term object, or empty (falsy).
 * @param bool  $ignore Whether the settings have been ignored.
 * @return array $data The new description data.
 */
add_filter( 'the_seo_framework_generated_description_additions', function( $data = array(), $id = 0, $term = '', $ignore = false ) {
	// Example 1: Change title for $id to 'Something' if $id is 5.
	$data['title'][ $id ] = 5 === $id ? 'Something' : $data['title'][ $id ];

	// Example 2: If $ignore is true, empty all known data values.
	if ( $ignore )
		$data['title'] = $data['on'] = $data['blogname'] = $data['sep'] = '';

	return $data;
}, 10, 4 );
```
